src/containers/PageBuilder/Primitives/SearchCTA/SearchCTA.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Field, Form as FinalForm } from 'react-final-form';
+import { stringify } from 'query-string';
+import { useHistory } from 'react-router-dom';
+import { createResourceLocatorString } from '../../../../util/routes';
+import { useRouteConfiguration } from '../../../../context/routeConfigurationContext';
+import { useConfiguration } from '../../../../context/configurationContext';
+
+import FilterCategories from './FilterCategories';
+import FilterDateRange from './FilterDateRange';
+import FilterLocation from './FilterLocation';
+import FilterKeyword from './FilterKeyword';
+import { Button, Form } from '../../../../components';
+
+import css from './SearchCTA.module.css';
+
+const GRID_CONFIG = [
+  { gridCss: css.gridCol1 },
+  { gridCss: css.gridCol2 },
+  { gridCss: css.gridCol3 },
+  { gridCss: css.gridCol4 },
+];
+
+const getGridCount = numberOfFields => {
+  const gridConfig = GRID_CONFIG[numberOfFields - 1];
+  return gridConfig ? gridConfig.gridCss : GRID_CONFIG[0].gridCss;
+};
+
+export const SearchCTA = React.forwardRef((props, ref) => {
+  const history = useHistory();
+  const routeConfiguration = useRouteConfiguration();
+  const { categories, dateRange, keywordSearch, locationSearch } = props.searchFields;
+
+  //  If no search fields are enabled, we return null (Console won't allow you to enable 0 search fields)
+  const noSearchFields = !(categories || dateRange || keywordSearch || locationSearch);
+  if (noSearchFields) {
+    return null;
+  }
+
+  const onSubmit = values => {
+    // Convert form values to query parameters
+    const queryParams = {};
+
+    // WIP
+    Object.entries(values).forEach(([key, value]) => {
+      if (value) {
```

Comment:
I think this should be more explicit. If someone customizes this to include numbers, we'll end up to the notorious _zero is falsy_ situation.

Something like:

```js
const isEmpty = value => {
  const isNullish = value == null;
  const isZeroLength = value.hasOwnProperty('length') && value.length === 0;
  return isNullish || isZeroLength;
}
```

Reply:
Perhaps also relevant for the noSearchField?
```js
const noSearchFields = isEmpty([categories, dateRange, keywordSearch, locationSearch].filter(f => !isEmpty(f)));
```